lib/src/neu/types.ts

```diff
@@ -270,14 +282,32 @@ export function isBooleanLiteralType(type: Type): type is BooleanLiteralType {
   return type.kind === TypeKind.BOOLEAN_LITERAL;
 }
 
+export function areBooleanLiteralTypes(
+  types: Type[]
+): types is BooleanLiteralType[] {
+  return types.every(isBooleanLiteralType);
+}
+
 export function isStringType(type: Type): type is StringType {
   return type.kind === TypeKind.STRING;
 }
 
+export function isNotStringType<T extends Type>(
```

Comment:
Why creating another function when you just need to negate `isStringType` ?

Reply:
Using the negation in `.filter`: `types.filter(t => !isStringType(t))`. Whilst this will do the job, `types` still has type `Type[]`. 

Using a type guard function directly: `types.filter(isNotStringType)` allows type narrowing to occur. Now `types` is `StringType[]`.

Reply:
But you don't have to duplicate the logic, right ? 
`export const isNotStringType = (type) => !isStringType(type)`
or, if using lodash 
`export const isNotStringType = _.negate(isStringType)`